docs/src/tutorials/symbmc.md

```diff
@@ -0,0 +1,655 @@
+  `[T -> BOOLEAN]` and `SUBSET T`: Each predicate `P \in [T -> BOOLEAN]` corresponds to the set `{ t \in T: P(t) = TRUE} \in SUBSET T`. For this reason, predicates are often directly identified with the subset they are equivalent to, and we write `P \subseteq T` for brevity.
+  - Infix notation: Given a relation `R \in [A \X B -> BOOLEAN]`, we commonly write `a R b` instead of `R(a,b)` (e.g. `a > b` instead of `>(a,b)`).
+  - Cartesian product: Given a set `T`, we use `T^2` to refer to `T \X T`. `T^k`, for `k > 2` is defined similarly.
+  - Common sets: We use the shorthand `Z` to refer to the set of all integers, and `N` to refer to the set of all naturals, i.e. `N = {z \in Z: z >= 0}`.
+
+## On state-spaces and transition systems
+A TLA+ specification defines a triple `(S,S0,->)`, called a _transition system_. `S` is the _state space_, `S0` is the set of initial states (`S0 \subseteq S`), and `->` is the transition relation, a subset of `S^2`.
+
+### State spaces
+The notion of a single state depends on the specification, in particular, on the (number of) variables it declares. If a specification declares 
+```tla
+VARIABLE A1, A2, A3, ..., Ak
+```
+then a _state_ is a `k`-field record `[A1 |-> a1, ..., Ak |-> ak]`, where `ai` represents the value of the variable `Ai`, for each `i = 1,...,k`. The state space `S` is then the set of all such `k`-records, i.e. the set of all possible combinations of values that variables may hold.
+For brevity, whenever the specification defines exactly one variable, we will treat a state as a single value `a1` instead of the record `[A1 |-> a1]`.
+
+In untyped TLA+, one can think of `S` as `[A1:U, ..., Ak: U]`, which is naturally isomorphic to `U^k`, where `U` is the universe of all TLA+ values. In typed TLA+, such as in Apalache, where variable declarations look like:
+```tla
+VARIABLE 
+  \* @type: T1;
+  A1, 
+  ..., 
+  \* @type: Tk;
+  Ak
+```
+ `S` is the set `[A1: V1, ..., Ak: Vk]`, where `Vi` is the set of all values, which hold the type `Ti`, for each `i = 1,...,k`.
+For example, in the specification with
+```tla
+VARIABLE 
+  \* @type: Bool;
+  a, 
+  \* @type: Bool;
+  b
+```
+The state space is `[x: BOOLEAN,y: BOOLEAN]` when considering types, since each variable can hold one of two boolean values. In the untyped setting, the state space is infinite, and contains all pairs of TLA+ values, so, for example
+```
+{
+  [ x |-> 1, 
+    y |-> TRUE ], 
+  [ x |-> [z \in 1..5 |-> "a"], 
+    y |-> CHOOSE p \in {}: TRUE ]
+} \subset S
+```
+
+As Apalache enforces a type system,the remainder of this document will assume the typed setting. This assumption does not change any of the definitions. We will also assume that every specification declares an initial-state predicate `Init`, a transition-predicate `Next` and an invariant `Inv` (if not specified, assumed to be `TRUE`). For simplicity, we will also assume that the specification if free of constants, resp. that all of the constants have been initialized.
+
```

Comment:
I think it matters, because it changes what the definition of `S` is. It wouldn't be strange for someone to expect the state space to be its untyped variant, if I didn't explicitly note the differences between the two. There is no such thing as "typed TLA+" in the community, be default.

Reply:
But if you link to where we introduce it, they could navigate there, and you'd save a good chunk of words, so readers could focus on the key details.